platform/mellanox/onie-fw-update

```diff
@@ -0,0 +1,140 @@
+#!/bin/sh
+
+# Copyright (C) 2019 Mellanox Technologies Ltd.
+# Copyright (C) 2019 Michael Shych <michaelsh@mellanox.com>
+#
+# SPDX-License-Identifier:     GPL-2.0
+
+this_script=${ONIE_FWPKG_PROGRAM_NAME:-$(basename $(realpath $0))}
+
+onie_mount=/mnt/onie-boot
+os_boot=/host
+onie_partition=
+onie_entry=0
+
+export ONIE_FWPKG_PROGRAM_NAME=$(basename $(realpath $0))
+
+usage()
+{
+cat <<EOF
+  update
+    The 'update' command will reboot system to ONIE update mode
+    and ONIE will perform automatically update of previously 
+    added (i.e. pending) FW (ONIE itself, BIOS or CPLD) image.
+
+EOF
+}
+
+enable_onie_access()
+{
+	onie_partition=$(fdisk -l | grep "ONIE boot" | awk '{print $1}')
+	if [ ! -d $onie_mount ]; then
+		mkdir /mnt/onie-boot
+	fi
+	mount $onie_partition /mnt/onie-boot
+	if [ ! -e /lib/onie ]; then
+		ln -s /mnt/onie-boot/onie/tools/lib/onie /lib/onie
+	fi
+	PATH=/sbin:/usr/sbin:/bin:/usr/bin:$onie_mount/onie/tools/bin/
+	export PATH
+}
+
+clean_onie_access()
+{
+	rm -f /lib/onie
+	umount $onie_partition
+}
+
+# ONIE entry must exist in grub config
+find_onie_menuentry()
```

Comment:
@stephenxs Since we are already use `onie_entry`, we do not need this anymore

Reply:
removed.